tests/unit/test_upload.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for upload functionality"""
+
+
+import pytest
+
+from encryption_key_store.core.envelope_decryption import extract_envelope_content
+
+from ..fixtures.file_fixture import first_part_fixture  # noqa: F401
+from ..fixtures.file_fixture import FirstPartFixture
+from ..fixtures.ghga_secrets import generate_secrets_fixture  # noqa: F401
+from ..fixtures.ghga_secrets import ghga_secrets_dao_fixture  # noqa: F401
+
+
+@pytest.mark.asyncio
+async def test_extract(
```

Comment:
Great that it is here. Just wanted to mention that we don't require unit tests in this prototyping phase. Only write them if they help you sketch the code (if you have to write them after all the features are complete, we can skip them for now).